packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -2,6 +2,105 @@ import { AccessType, AccessAction } from './grant'
 import { PaymentPointerContext } from '../../app'
 import { Transaction } from 'objection'
 import { GrantReference } from '../grantReference/model'
+import {
+  createVerifier,
+  httpis,
+  RequestLike,
+  Algorithm as AlgorithmName
+} from 'http-message-signatures'
+import { Request as KoaRequest } from 'koa'
+import { JWKWithRequired } from 'auth'
+import {
+  ByteSequence,
+  InnerList,
+  Item,
+  parseDictionary,
+  serializeDictionary
+} from 'structured-headers'
+
+function parseAlgorithmName(alg: string): AlgorithmName {
+  if (alg === 'EdDSA' || alg === 'ed25519') {
+    // ed25519 is EdDSA
+    return 'ed25519'
+  } else {
+    throw new Error(
+      `The signature parameter 'alg' is using an illegal value '${alg}'. Only 'ed25519' ('EdDSA') is supported.`
+    )
+  }
+}
+
+async function verifyRequest(
+  koaRequest: KoaRequest,
+  jwk: JWKWithRequired,
+  key: string
+): Promise<void> {
+  const { kid, kty } = jwk
+
+  if (kty !== 'OKP') {
+    throw new Error('invalid key type')
+  }
+
+  if (!koaRequest.headers['signature']) {
+    throw new Error('signature is missing')
+  }
+
+  if (!koaRequest.headers['signature-input']) {
+    throw new Error('signature-input is missing')
+  }
+
+  const signatureInputMap = new Map<string, string>()
+  signatureInputMap.set('keyid', kid)
+  signatureInputMap.set('alg', parseAlgorithmName(jwk.alg))
```

Comment:
`signatureInputMap` isn't used anywhere.
Was it intended to be added to `typedRequest`?

re: `alg`
I'm trying to decide if we should either:
- update our `http-message-signatures` to not expect `alg` in the `Signature-Input` header given that:
>   If the signing algorithm is a JOSE signing algorithm from the JSON
   Web Signature and Encryption Algorithms Registry established by
   [RFC7518], the JWS algorithm definition determines the signature and
   hashing algorithms to apply for both signing and verification.  There
   is no use of the explicit "alg" signature parameter when using JOSE
   signing algorithms.

https://datatracker.ietf.org/doc/html/draft-ietf-httpbis-message-signatures-06#section-3.3.5
- not use `http-message-signatures` for now, and just `export`/`import` the AS's [httpsig verification](https://github.com/interledger/rafiki/blob/6cb56141013d38197b13b31e392ebbee402adcd6/packages/auth/src/signature/service.ts)

Reply:
For the purposes of this PR, I just did the first option and removed the check for `alg` in the `Signature-Input` header